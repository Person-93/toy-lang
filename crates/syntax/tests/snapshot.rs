use anyhow::{Context, Result};
use chumsky::{Parser, Stream};
use insta::{assert_debug_snapshot, with_settings};
use logos::Logos;
use std::fs;

fn main() -> Result<()> {
  let dir = concat!(env!("CARGO_MANIFEST_DIR"), "/../../toys");
  for entry in fs::read_dir(dir).with_context(|| format!("missing toys directory: {dir}"))? {
    let entry = entry.context("failed reading directory entry")?;
    if !entry
      .file_type()
      .context("failed reading file type")?
      .is_file()
    {
      continue;
    }
    let path = entry.path();
    if let Some(ext) = path.extension() {
      if ext.to_str().unwrap() != "toy" {
        continue;
      }
    }

    let text = fs::read_to_string(&path)
      .context("failed reading file")?
      // strip '\r' so snapshots match on windows
      .replace('\r', "");
    let tokens = toy_lang_syntax::tokens::Token::lexer(&text).spanned();
    with_settings!({snapshot_suffix => path.file_name().unwrap().to_str().unwrap()}, {
      assert_debug_snapshot!(toy_lang_syntax::ast::parse::file().parse_recovery_verbose(Stream::from_iter(text.len()..text.len(), tokens)));
    });
  }

  Ok(())
}
