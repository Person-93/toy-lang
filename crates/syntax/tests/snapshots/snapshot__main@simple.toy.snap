---
source: crates/syntax/tests/snapshot.rs
assertion_line: 31
expression: "toy_lang_syntax::ast::parse::file().parse_recovery_verbose(Stream::from_iter(text.len()..text.len(),\n                                                                             tokens))"

---
(
    Some(
        [
            TopLevel {
                vis: None,
                item: Function(
                    Function {
                        extern_: None,
                        ident: Ident(
                            "f",
                        ),
                        generic_params: None,
                        params: [],
                        return_type: None,
                        body: Some(
                            Expr {
                                primary: CodeBlock(
                                    CodeBlock {
                                        statement: [],
                                        expr: None,
                                    },
                                ),
                                args: None,
                            },
                        ),
                    },
                ),
            },
        ],
    ),
    [],
)
